tardis/adapters/sites/slurm.py

```diff
@@ -0,0 +1,131 @@
+                                                                      FAILED=ResourceStatus.Error,
+                                                                      NODE_FAIL=ResourceStatus.Error,
+                                                                      OUT_OF_MEMORY=ResourceStatus.Error,
+                                                                      PENDING=ResourceStatus.Booting,
+                                                                      RUNNING=ResourceStatus.Running,
+                                                                      REQUEUED=ResourceStatus.Error,
+                                                                      RESIZING=ResourceStatus.Error,
+                                                                      REVOKED=ResourceStatus.Error,
+                                                                      SUSPENDED=ResourceStatus.Running,
+                                                                      TIMEOUT=ResourceStatus.Stopped):
+                                             translator[x],
+                                             JobId=lambda x: int(x))
+
+        self.handle_response = partial(self.handle_response, key_translator=key_translator,
+                                       translator_functions=translator_functions)
+
+    async def deploy_resource(self, resource_attributes):
+        request_command = f'sbatch -p {self.configuration.MachineTypeConfiguration[self._machine_type].Partition} ' \
+                          f'-N 1 -n {self.machine_meta_data.Cores} ' \
+                          f'--mem={self.machine_meta_data.Memory}gb ' \
+                          f'-t {self.configuration.MachineTypeConfiguration[self._machine_type].Walltime} ' \
+                          f'{self._startup_command}'
+        result = await self._executor.run_command(request_command)
+        logging.debug(f"{self.site_name} servers create returned {result}")
+        pattern = re.compile(r'^Submitted batch job (\d*)', flags=re.MULTILINE)
+        remote_resource_uuid = int(pattern.findall(result.stdout)[0])
+        resource_attributes.update(remote_resource_uuid=remote_resource_uuid, created=datetime.now(),
+                                   updated=datetime.now(), drone_uuid=self.drone_uuid(remote_resource_uuid),
+                                   resource_status=ResourceStatus.Booting)
+        return resource_attributes
+
+    @property
+    def machine_meta_data(self):
+        return self.configuration.MachineMetaData[self._machine_type]
+
+    @property
+    def machine_type(self):
+        return self._machine_type
+
+    @property
+    def site_name(self):
+        return self._site_name
+
+    async def resource_status(self, resource_attributes):
+        await self._slurm_status.update_status()
+        resource_status = self._slurm_status[str(resource_attributes.remote_resource_uuid)]
```

Comment:
What will happen, if the `remote_resource_uuid` is not yet part of the dictionary. Since the update is done asynchronous, it can happen.